src/mjolnir/osmdata.cc

```diff
@@ -525,6 +562,29 @@ bool read_linguistic(const std::string& filename, LinguisticMultiMap& ling_map)
   return true;
 }
 
+bool read_conditional_speed_limits(const std::string& filename,
+                                   ConditionalSpeedLimitsMultiMap& speed_map) {
+  // Open file and truncate
+  std::ifstream file(filename, std::ios::in | std::ios::binary);
+  if (!file.is_open()) {
+    LOG_ERROR("read_conditional_speed_limits failed to open input file: " + filename);
+    return false;
+  }
+
+  // Read the count and then the temporary access restriction list
+  uint32_t count = 0;
+  file.read(reinterpret_cast<char*>(&count), sizeof(uint32_t));
+  std::vector<TempConditionalSpeedLimit> flat(count);
```

Comment:
Why not skip making the `flat` vector and emplace into the map right away with a loop?

Reply:
Well, all osmdata items are stored via temp flat structure and via flat vector. I definitely don't want to refactor whole osmdata.cc in this PR. Probably there were a reasons for such approach that I don't know

Reply:
Addressed in b08327ce2b9c8cbbbf0ac0bd8826a39f096baa17